src/shell/interpreter.zig

```diff
@@ -4241,10 +4301,16 @@ pub const Interpreter = struct {
                             cmd.writeFailingError("bun: ambiguous redirect: at `{s}`\n", .{@tagName(kind)});
                             return .yield;
                         }
+
+                        // Regular files are not pollable on linux
+                        const is_pollable: bool = if (bun.Environment.isLinux) false else true;
+
                         const path = cmd.redirection_file.items[0..cmd.redirection_file.items.len -| 1 :0];
                         log("EXPANDED REDIRECT: {s}\n", .{cmd.redirection_file.items[0..]});
                         const perm = 0o666;
-                        const flags = node.redirect.toFlags();
+                        const is_nonblocking = (node.redirect.stdout or node.redirect.stderr) and is_pollable;
+                        var flags = node.redirect.toFlags();
+                        if (is_nonblocking) flags |= std.os.O.NONBLOCK;
```

Comment:
stdout or stderr are likely blocking. If we open them as non-blocking, then it will cause bugs with other processes.